lib/hamster/hash.rb

```diff
@@ -590,6 +590,27 @@ def fetch_values(*wanted)
       Vector.new(array.freeze)
     end
 
+    # Return the value of successively indexing into a hash
+    # If any of the keys is not present in the hash, return `hil`
+    #
+    # @example
+    #   h = Hamster::Hash[:a => 9, :b => Hamster::Hash[:c => 'a', :d => 4], :e => nil]
+    #   h.dig(:b, :c) # => "a"
+    #   h.dig(:b, :f) # => nil
+    # @params keys to fetch from the hash
+    # @return value
+    def dig(key, *rest)
+      value = self[key]
+      if rest.empty? || value.nil?
+        value
+      else
+        if value.respond_to?(:dig)
+          value.dig(*rest)
+        end
+      end
+    end
```

Comment:
Sorry, I should have been more clear. There seems to some debate if `Array#dig` should raise an error when indexed with a non integer, or return `nil`. Which do you think fits better with existing hamster apis?

Sounds like you are saying raising an error is expected?